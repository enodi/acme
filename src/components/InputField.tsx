import React from "react";

const InputField: React.FC = () => <h1>InputField</h1>;

export default InputField;
