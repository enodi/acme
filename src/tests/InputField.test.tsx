import React from "react";
import { shallow } from "enzyme";

import InputField from "../components/InputField";
import { selectedItem } from "./fixtures";

let handleKeyDown,
  onTextChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
  wrapper: any;
beforeEach(() => {
  handleKeyDown = jest.fn();
  onTextChange = jest.fn();
  wrapper = shallow(
    <InputField
      handleKeyDown={handleKeyDown}
      onTextChange={onTextChange}
      messageText={selectedItem.messages[0]}
    />
  );
});

test("should render InputField component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should handle text change", () => {
  const value = selectedItem.messages[0];
  wrapper.find("input").simulate("change", value);
  expect(onTextChange).toHaveBeenLastCalledWith(value);
});
