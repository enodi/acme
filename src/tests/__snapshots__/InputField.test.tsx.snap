// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render InputField component correctly 1`] = `
<input
  className="input-field"
  onChange={[MockFunction]}
  onKeyDown={[MockFunction]}
  placeholder="Send a message"
  value="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
/>
`;
